bills-app/src/main/webapp/script.js

```diff
@@ -818,3 +820,52 @@ const renderBills = (bills) => {
         billsList.appendChild(billsListItem);
     });
 }
+
+function serchBillsWithAPI() {
+    var searchquery = document.getElementById("searchbutton").value;
+
+    if (searchquery.length != 0){
+        //resultLimit specifies the number of bills to be returned to the user. This number cannot exceed 20.
+        var resultLimit = 5;
+
+        var url="https://api.propublica.org/congress/v1/bills/search.json?query=" + searchquery;
+        fetch(url, {method: "GET", headers: {"X-API-Key": proPublicaApiKey}
+        }).then(response => response.json()).then((searchedBills) => {
+            //searchedBills is the returned json file
+            var billsList = searchedBills.results[0].bills;
+            document.getElementById('API-search-container').innerText = billsToString(billsList, resultLimit);
+        });
+    } else { //if the user clicks on submit without a query
+        document.getElementById('API-search-container').innerText = "";
+    }
+}
+
+function billsToString(billsList, limit) {
+    resultsString = "";
+
+    if (billsList.length < limit){
+        limit = billsList.length;
+    }
+    
+    if(billsList.length == 0) {
+        resultsString = "No results found.";
+    }
+    else {
+        for (var i = 0; i<limit; i++) {
```

Comment:
Really like this part of the code! I think there could be a cleaner way of iterating through the billsList array with the `.map()` method. Great code overall :)